modules/integration/tests/src/test/java/org/wso2/carbon/am/tests/sample/APIScopeTestCase.java

```diff
@@ -0,0 +1,239 @@
+/*
+*  Copyright (c) 2015, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+*
+*  WSO2 Inc. licenses this file to you under the Apache License,
+*  Version 2.0 (the "License"); you may not use this file except
+*  in compliance with the License.
+*  You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing,
+* software distributed under the License is distributed on an
+* "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+* KIND, either express or implied.  See the License for the
+* specific language governing permissions and limitations
+* under the License.
+*/
+package org.wso2.carbon.am.tests.sample;
+
+import org.json.JSONObject;
+import org.testng.Assert;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+import org.wso2.carbon.am.tests.APIManagerIntegrationTest;
+import org.wso2.carbon.am.tests.util.APIPublisherRestClient;
+import org.wso2.carbon.am.tests.util.APIStoreRestClient;
+import org.wso2.carbon.am.tests.util.bean.APILifeCycleState;
+import org.wso2.carbon.am.tests.util.bean.APILifeCycleStateRequest;
+import org.wso2.carbon.am.tests.util.bean.APIRequest;
+import org.wso2.carbon.am.tests.util.bean.GenerateAppKeyRequest;
+import org.wso2.carbon.am.tests.util.bean.SubscriptionRequest;
+import org.wso2.carbon.automation.api.clients.user.mgt.UserManagementClient;
+import org.wso2.carbon.automation.core.utils.HttpRequestUtil;
+import org.wso2.carbon.automation.core.utils.HttpResponse;
+
+import java.net.URL;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * This covers adding two scopes with different roles and accessing them with users with those roles
+ * This Test-case uses the existing admin role and admin user
+ * Doc For API used :- http://sqlrest.sourceforge.net/5-minutes-guide.htm
+ */
+public class APIScopeTestCase extends APIManagerIntegrationTest {
+
+    private APIPublisherRestClient apiPublisher;
+
+    private APIStoreRestClient apiStore;
+
+    private UserManagementClient userManagementClient = null;
+
+    private static final String API_NAME = "APIScopeTestAPI";
+
+    private static final String API_VERSION = "1.0.0";
+
+    private static final String API_PROVIDER = "admin";
+
+    private static final String APP_NAME = "NewApplication";
+
+    private static final String USER_JOHN = "john";
+
+    private static final String SUBSCRIBER_ROLE = "subscriber";
```

Comment:
Currently our code is in frozen state due to release i will fix this once its released